packages/components/src/utils/table/table.test.ts

```diff
@@ -0,0 +1,136 @@
+import type { RowSelectionState, Table, Updater } from '@tanstack/react-table';
+import { createTable, getCoreRowModel } from '@tanstack/react-table';
+import { beforeEach, describe, expect, test } from 'vitest';
+import { addRow, deleteFirstSelectedRow, getFirstSelectedRow, selectRow } from './table';
+import type { TestData } from './test-utils/types';
+
+let data: Array<TestData>;
+let newRowData: TestData;
+
+let table: Table<TestData>;
+let onRowSelectionChangeValue: Updater<RowSelectionState>;
+
+beforeEach(() => {
+  data = [
+    { name: 'NameData0', value: 'ValueData0' },
+    { name: 'NameData1', value: 'ValueData1' },
+    { name: 'NameData2', value: 'ValueData2' }
+  ];
+  newRowData = { name: 'newDataName', value: 'newDataValue' };
+
+  table = createTable({
+    columns: [],
+    data: data,
+    getCoreRowModel: getCoreRowModel(),
+    onStateChange: () => {},
+    onRowSelectionChange: (value: Updater<RowSelectionState>) => {
+      onRowSelectionChangeValue = value;
+    },
+    renderFallbackValue: undefined,
+    state: {}
+  });
+  onRowSelectionChangeValue = {};
```

Comment:
also instead of hold this in a global let, better create a setup function which you call at the start of each test.

const { table, onRowSelectionChangeValue } = setupTable()